### ISSUE TITLE
v1.2x nodeconfig json not recognised by v2 beta

### ISSUE THREAD

**Issue by JohnOSu** [appium/appium#16593]
## The problem

I'm trying to launch a v2 server as a selenium grid node

`appium --log-timestamp --address 192.168.0.228 --port 7100  --nodeconfig /Users/automation/config/nodeconfig-ios-lgc.json --session-override`

I get this error:
`appium server: error: argument --nodeconfig: invalid json value: '/Users/automation/config/nodeconfig-ios-lgc.json'`

This json file works fine for 1.2x.

## Environment

* Appium version (or git revision) that exhibits the issue: 2.0.0-beta.25
* Last Appium version that did not exhibit the issue (if applicable): 1.21.0
* Desktop OS/version used to run Appium: MacOS 12.3
* Node.js version (unless using Appium.app|exe): v17.7.1
* Npm or Yarn package manager: npm
* Mobile platform/version under test: n/a
* Real device or emulator/simulator: n/a
* Appium CLI or Appium.app|exe: CLI

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium logs

Create a [GIST](https://gist.github.com) which is a paste of your _full_ Appium logs, and link them here.
Do _NOT_ paste your full Appium logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include Appium logs!

## Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

Please git clone https://github.com/appium/appium and from the `sample-code` directory, use one of your favourite languages and sample apps to reproduce the issue.

In case a similar scenario is missing in sample-code, please submit a PR with one of the sample apps provided.

**Comment by jlipps** [comment]
@boneskull this was one of those args that could be a json string or a path to a json file. how are we handling parsing those now?

**Comment by boneskull** [comment]
Appium doesn't support external config files.  if there's a good reason to keep this one as an external config file, then I can work to add support.  the new way to do it would be to create e.g., `.appiumrc.json` and add this to it:

```js
{
  "server": {
    "nodeconfig": <insert contents of yr .json file here>
  }
}
```

**Comment by boneskull** [comment]
in this case, if those config files are actually part of some other system's configuration, then I think it'd make sense to allow them to be read directly from disk.

**Comment by boneskull** [comment]
right, it basically just needs to be in the config file.  I had hoped to eliminate the need for external config files.  the only reason to keep it would be if the `.json` files are also used by selenium somehow.  is that the case?

**Comment by jlipps** [comment]
oh right, sorry, i forgot we can now specify a global appiumconfig and easily include these. so yes, the current state is fine, we should just have a section in the migration doc that shows how to do this.

**Comment by jlipps** [comment]
nope! we just need an explainer we can point people to for how to include this info in the config file.

**Comment by boneskull** [comment]
See #16661

**Comment by mykola-mokhnach** [comment]
Closed as obsolete
